pkg/kubelet/dockertools/manager.go

```diff
@@ -476,6 +476,14 @@ func (dm *DockerManager) GetPodStatus(pod *api.Pod) (*api.PodStatus, error) {
 		if containerStatus, found := statuses[container.Name]; found {
 			reason, ok := dm.reasonCache.Get(uid, container.Name)
 			if ok && reason == kubecontainer.ErrCrashLoopBackOff.Error() {
+				// We need to increment the restart count if we are going to
+				// move the current state to last terminated state.
+				if containerStatus.State.Terminated != nil {
```

Comment:
We might if we don't persist it to the apiserver.

This part of the code is really messy, and we should clean it up sometime.